adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -293,22 +293,27 @@ func (h *hnsw) reassignNeighborsOf(deleteList helpers.AllowList, breakCleanUpTom
 	h.RLock()
 	size := len(h.nodes)
 	h.RUnlock()
+	h.resetLock.Lock()
+	defer h.resetLock.Unlock()
 
-	for n := 0; n < size; n++ {
-		if ok, err := h.reassignNeighbor(uint64(n), deleteList, breakCleanUpTombstonedNodes); err != nil {
-			return false, errors.Wrap(err, "reassign neighbor edges")
-		} else if !ok {
-			return false, nil
+	compressionhelpers.Concurrently(uint64(size), func(deletedID uint64) {
```

Comment:
Should we add an environment variable to limit concurrency here?
Is 25% a better default than 100% total cores?